src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkMigrationService.kt

```diff
@@ -0,0 +1,103 @@
+
+    log.info("Migrating video link booking ID $videoBookingId")
+
+    val appointments = videoLinkAppointmentRepository.findAllByVideoLinkBooking(videoLinkBooking)
+    require(appointments.isNotEmpty()) { "Video link booking ID $videoBookingId has no appointments" }
+
+    // Extract the related appointments
+    val main = appointments.find { it.hearingType == HearingType.MAIN }
+    val pre = appointments.find { it.hearingType == HearingType.PRE }
+    val post = appointments.find { it.hearingType == HearingType.POST }
+
+    // Fail if the main appointment is not present
+    require(main != null) { "Video link booking ID $videoBookingId has no main appointment" }
+
+    // Get the history of events for this booking
+    val events = videoLinkBookingEventRepository.findEventsByVideoLinkBookingId(videoBookingId)
+    require(events.isNotEmpty()) { "Video link booking ID $videoBookingId has no events" }
+
+    return VideoBookingMigrateResponse(
+      videoBookingId = videoLinkBooking.id!!,
+      offenderBookingId = videoLinkBooking.offenderBookingId,
+      courtCode = videoLinkBooking.courtId,
+      courtName = videoLinkBooking.courtName,
+      madeByTheCourt = videoLinkBooking.madeByTheCourt ?: false,
+      prisonCode = videoLinkBooking.prisonId,
+      probation = videoLinkBooking.courtName?.contains("Probation") ?: false,
+      comment = videoLinkBooking.comment,
+      pre = pre?.let { mapAppointment(pre) },
+      main = mapAppointment(main),
+      post = post?.let { mapAppointment(post) },
+      events = mapEvents(events),
+    )
+  }
+
+  fun mapAppointment(appointment: VideoLinkAppointment) =
+    AppointmentLocationTimeSlot(
+      locationId = appointment.locationId,
+      date = LocalDate.from(appointment.startDateTime),
+      startTime = LocalTime.from(appointment.startDateTime).withSecond(0).withNano(0),
+      endTime = LocalTime.from(appointment.endDateTime).withSecond(0).withNano(0),
+    )
+
+  fun mapEventToLocationTimeSlot(locationId: Long, startTime: LocalDateTime, endTime: LocalDateTime) =
+    AppointmentLocationTimeSlot(
+      locationId = locationId,
+      date = LocalDate.of(startTime.year, startTime.month, startTime.dayOfMonth),
+      startTime = LocalTime.of(startTime.hour, startTime.minute),
+      endTime = LocalTime.of(endTime.hour, endTime.minute),
+    )
+
+  private fun mapEvents(events: List<VideoLinkBookingEvent>): List<VideoBookingMigrateEvent> {
+    val response = events.map { event ->
+      VideoBookingMigrateEvent(
+        eventId = event.eventId!!,
+        eventTime = event.timestamp,
+        eventType = event.eventType,
+        createdByUsername = event.userId?.let { event.userId } ?: "MIGRATED",
+        prisonCode = event.agencyId?.let { event.agencyId } ?: "UNKNOWN",
+        courtCode = event.courtId?.let { event.courtId } ?: "UNKNOWN",
+        courtName = event.court,
+        madeByTheCourt = event.madeByTheCourt ?: false,
```

Comment:
Same. Do you need the `false` here? madeByTheCourt = `videoLinkBooking.madeByTheCourt ?: false,`

Reply:
The `madeByTheCourt` attribute in the entity is nullable - so could be [true|false|null] - so if null, I'm setting it to false. I've made this a bit clearer now.